### ISSUE TITLE
Fuckr Multi Processes (Sessions) / Constant Crash Issue, Uses Phantom Resources 

### ISSUE THREAD

**Issue by powline99** [tomlandia/fuckr#16]
@tomlandia the newest version 1.6.1 still leaves multiple threads open in the processes after it crashes. It crashes quite often (i believe Grindr is logging me out and crashing possibly a eclipse login server issue - like if you have multiple instances open of your profile) on my Windows 32bit laptop and I must open up a new instance which takes more resources... 

* Error Pops up every so often (while chatting, deleting, or blocking), eclipse wants login (I must go put in CAPTCHA phrase and click to get rid of Fuckr errors. Shows brief screen with unblocked local profiles. Second image is a error type thing after CAPTCHA phrase and then it goes to main screen.

![2016-04-28_082528](https://cloud.githubusercontent.com/assets/16879733/14893410/2af91562-0d24-11e6-8f77-cc0b97791d4f.jpg)

![2016-04-28_072714](https://cloud.githubusercontent.com/assets/16879733/14893412/30ca551e-0d24-11e6-9e16-dfcdcabd8554.jpg)

**Could we implement that "Looks like https://groups.google.com/forum/#!topic/nwjs-general/LtJql3LSV44 may offer a solution provided it's stable and we can add a nw command line flag to releases"**

___________________________________________________________________
Profile information Looking For / Tribe / Ethnicity / Body Type Missing in Profile Text

1) Profile information: Looking For / Tribe / Ethnicity / Body Type / relationship status Seems to be Missing from individual Profile Text in the table area with existing height/weight/age

2) *** Need drop down filter: ## "Relationship statuses" (ie. Single / open relationship)

3) Need drop down filter: height / weight preferences.

**Comment by victorgrego** [comment]
Thomas, what if there is a new view where we can do advanced filtering, just like grindr does. If u approve i can implement.

**Comment by tomlandia** [comment]
1) As Chrome is multi-process, I don't think there's much I can do about the multiple processes :/. Any idea why the app crashes on Windows could be helpful though

**Comment by powline99** [comment]
Hi Tomlandia Thanks for your response to followup: 

1) I cannot quit the processes as you can see by the screen shot, hitting end does nothing which is troubling that they are stuck, they still are stuck in memory until you do a restart. Note: "f.exe is Fuckr renamed and Fuckr.exe is another instance. See attached images below.
![image](https://cloud.githubusercontent.com/assets/16879733/12662650/c4a4741c-c5d6-11e5-82ff-9a982de18afa.png)
![image](https://cloud.githubusercontent.com/assets/16879733/12662694/ffdc43b6-c5d6-11e5-8db3-042fce58682f.png)
2) Guessing would be great for unit conversion: imperial / metric:  miles, weight & height.  Thank you!
3) That would be great to be able to change color of Top Pink Bar and Chat bubbles in units to keep things not so obvious that your Grinding 24-7 ;-P
4) Block right now is a 4+ step process which makes it hard since in Grindr blocking is used to keep those you are interested in and blocking those you aren't.  The process is: clicking individual profiles > then click chat icon > then click block > and hit ok to the popup dialog > which then sits there in your inbox after the block instead of returning to the tiled profile wall so you have to hit the 4 square icon to return (this should be automatic to the main view). A keyboard shortcut or less steps would be great and returning after a block to to the main view automatically.
5) Feature request: It would be great to get those advanced filters  for search

New Request Del Message in Inbox:
6 ) Grindr has to many spambots (mainly with short heights). Unlike the default app there is no way to hit "del" and remove individual messages from your inbox in Fuckr of those spambots. Thy could be blocked but I find they reappear and seem to contact again if you do not delete and ignore their messages. 

Thank You!

**Comment by genevera** [comment]
+1 for advanced filtering.  I was just about to fork this repository to add it in for myself.

**Comment by mratmsf** [comment]
I have convert units to feet, miles, lbs in my fork fuckr-us check it out.

**Comment by victorgrego** [comment]
1) About the multi processes. Looks like this only happens in win10. I've tested on mine and same thing happens. When I run on win7 its ok, but still crashes after some time.

2) have implemented here.

4)...

5) Idk whats happening. I send the request with the filtering parameters, but seems like it does not filter... maybe you could give me a hand @tomlandia 

6) I dont get it. I know those bots, but after blocked there may have another acc that contacts you. Whats your suggestion exactly.

**Comment by powline99** [comment]
In reply to @victorgrego:

Thanks for your reply...

1) I am on windows 7, there is a memory leak. I cannot use kill process it does nothing, even restart seems to be stuck. I included the process lists in photos above (prior posting). "f.exe" and "fuckr.exe" are two instances stuck in processes of the same program.

2) Thank You!
4) ...
5) ...
6) I am asking for a way to delete messages from my inbox. On a app you are able to swipe and select delete. If a bot contacts me I never block as they create another account or are notified and contact you right back. Same thing with those group orgy scammers who want information, It's best to ignore them and delete.

**Comment by Kav28** [comment]
I'm on windows 10 and all I get is one fuckr under apps, 3 fuckr under
background processes.  Logging in/out without restarting and restating had
no effect - always had the same number.

Think what hes asking in regards to "delete" is that from the chats window
there would be an edit button that would allow you to select multiple chats
to delete - ie clean up old chats with multiple users. This exists in the
iPad version but not Android version (of grindr)

On Sun, Jan 31, 2016 at 2:29 PM, Victor Nascimento <notifications@github.com

**Comment by powline99** [comment]
New Request:
7) Emoji icons, so many profiles use them instead of text it's hard to see their usage in profiles and then the person your chatting with acts like your stupid because you could not see their Emoji.

**Comment by victorgrego** [comment]
1) Okay, sorry @powline99 no clue on that, in my laptops it works just fine.
2) Tom has implemented.
3) I ill leave this to Tom, as my first try to change the visuals wasn't accepted :stuck_out_tongue: 
4) **Still Open**
5) I got to filter most settings. Surprisingly, Grindr does not limit to filter other settings via code. However, I did not figure out how to filter by: "Relationship Status" nor "Grindr Tribes", the others work just fine.
6) **Still Open**, maybe will take time.
7) **Still Open**, I found some plugins on that, but still did not test.

I'm pushing my **very partial** solutions hoping for the feedback of you guys :smile:

**Comment by powline99** [comment]
Thank you @victorgrego  some updated points

1) It is so annoying I am still getting crashes on Windows 7, Do I send you guys a app log or something? I must restart, restart , .... The memory leak is quite big with the resources it it taking staying in processes and no way to kill it or restart. 
2) Would love to get the Updated US Conversions Imperial is it compared to Metric 
3) Yes I love Scruffs Minimal Darker Design (no pink), is there a Scruff PC Clone to Use haha ;-p
4) The block issue is getting annoying especially since Grindr losts all my blocks and now everyone is unblocked with their recent issues. I have to reblock a ton of people. Why can't we be on a profile and use a keyboard shortcut Cntrl + b or whatever. It's a 4 step process that leaves you in your inbox when done currently not back in the main wall.
5) I asked for the ability to filter by ethnicity, tribe, relationship status, orientation, etc... I also need the ability to see that in everyone's profile, these stats are missing ... I believe it's just height/weight/age.
6) Delete messages from inbox would be great needing this function more and more.
7) Need emoji support, people think you are mentally challenged when you cannot understand the emoji icons you never saw, haha.
8) New Feature request: Why are pictures so small people send you, there is no way to click and see them in a bigger size, one must save each and every pic and use a window software.
9) Possible bug, why am I getting crashes when I goto my inbox and click on someone's photo so I can see their Grindr Profile it crashes the app. Have they blocked me and yet the chat log is stuck in my inbox still? There is no way to view their profile just the chat saying hey and the preview image attached to the inbox log. 

Thanks @victorgrego  @tomlandia

**Comment by victorgrego** [comment]
About the isse 5) I've implemented in last push :D
8) I've done myself a all pictures a link, so i can click and open in navigator.

**Comment by powline99** [comment]
@victorgrego 

How do I download your fork? Do I just pull the files to the existing exe folder or how do I get the Win 32 bit download
Thanks!

**Comment by Kav28** [comment]
Think (9) happens because the profile was deleted. Think what happens when
you have a chat is that the info is stored locally and when you click on
the profile the local info is updated (have seen an "old" pic change and
when I checked using the app the new pic was the profile pic).

(8) pics are small but when I click on them I get large images

(1) check to see if your java needs updating

Dennis

**Comment by powline99** [comment]
@Kav28 

Thanks Kav for your help

I updated java, the problem being I cannot delete Fuckr because it says it is in use and it will not let me kill the process. I have tried restarting. I use chromium as a web browser and it does not support IMAPI plugins. 

I found this is  the event viewer if anyone @victorgrego @tomlandia or @Kav28 can make since of it and help it would be greatly appreciated

Event 1000, Application Error
Event ID: 1000
Task Category: (100)
Faulting application name: Fuckr.exe, version: 0.0.0.0, time stamp: 0x552a925b
Faulting module name: Fuckr.exe, version: 0.0.0.0, time stamp: 0x552a925b
Exception code: 0x80000003
Fault offset: 0x000ec689
Faulting process id: 0x21d4
Faulting application start time: 0x01d156d1605ee41b
Faulting application path: D:\Fuckr-win32\Fuckr.exe
Faulting module path: D:\Fuckr-win32\Fuckr.exe


   [ Name]  Application Error 
 
  - EventID 1000 

   [ Qualifiers]  0 
 
   Level 2 
 
   Task 100 
 
   Keywords 0x80000000000000 
 
- EventData 

   Fuckr.exe 
   0.0.0.0 
   552a925b 
   Fuckr.exe 
   0.0.0.0 
   552a925b 
   80000003 
   000ec689 
   21d4 
   01d156d1605ee41b 
   D:\Fuckr-win32\Fuckr.exe 
   D:\Fuckr-win32\Fuckr.exe

**Comment by powline99** [comment]
@mratmsf @victorgrego Help guys... How do I combine the Win 32bit Fuckr Install with your forked versions? Say I want Victors Updates of the profile specs he did recently, and I want mratmsf's US Imperial conversions of miles /height/weight and I have Tom's original 32bit Install. 
Thanks!

**Comment by Kav28** [comment]
How do you normally use fuckr?  do you have multiple on going chats or just
one at a time? do you leave it running between sessions of use or sign out?

**Comment by powline99** [comment]
@Kav28 I leave it running if I am on my Laptop, when I go mobile I use the app. I have 2 instances of the app. I rarely sign out of Grindr. I can't figure out how to use the forked updates people have made to the 32 bit Windows install from the fuckr website, do I have to wait till the official release is updated or it their a way to use the Github forks.

**Comment by Kav28** [comment]
I suspect that if you have an active chat a process is forked to handle
that chat and stays active till you sign out of fuckr - many active chats
will lead to many fuckrs.  If you sign out of fuckr (which is not an option
in the app) the processes get cleaned up. By rarely signing out you keep
these processes active and they will build up - a testable theory at least
(I'm not observing what you are but I'm on windows 10 and dont keep the app
up).

By instances do you mean two grindr accounts running at the same time, two
grindr accounts but only one running at a time or two grindr accounts
running at the same time?

The files changed by any fix are listed in these updates. If I want the
change before a release I download the diff files and change the code as
indicated.

**Comment by Kav28** [comment]
You mentioned earlier about a restart of Windows not helping.  You should do a shutdown/startup not a restart.  Can be different. 

On Feb 5, 2016, at 8:28 PM, powline99 <notifications@github.com> wrote:

@Kav28

Thanks Kav for your help

I updated java, the problem being I cannot delete Fuckr because it says it is in use and it will not let me kill the process. I have tried restarting. I use chromium as a web browser and it does not support IMAPI plugins.

I found this is the event viewer if anyone @victorgrego @tomlandia or @Kav28 can make since of it and help it would be greatly appreciated

Event 1000, Application Error
Event ID: 1000
Task Category: (100)
Faulting application name: Fuckr.exe, version: 0.0.0.0, time stamp: 0x552a925b
Faulting module name: Fuckr.exe, version: 0.0.0.0, time stamp: 0x552a925b
Exception code: 0x80000003
Fault offset: 0x000ec689
Faulting process id: 0x21d4
Faulting application start time: 0x01d156d1605ee41b
Faulting application path: D:\Fuckr-win32\Fuckr.exe
Faulting module path: D:\Fuckr-win32\Fuckr.exe

[ Name] Application Error

EventID 1000

[ Qualifiers] 0

Level 2

Task 100

Keywords 0x80000000000000

EventData

Fuckr.exe 
0.0.0.0 
552a925b 
Fuckr.exe 
0.0.0.0 
552a925b 
80000003 
000ec689 
21d4 
01d156d1605ee41b 
D:\Fuckr-win32\Fuckr.exe 
D:\Fuckr-win32\Fuckr.exe

**Comment by powline99** [comment]
@Kav28 that could be a highly possible theory. So help me understand this. Every time I chat with multiple people on Fuckr and I do not logout there are forked resources which acts as basically active tabs in a web browser and use resources. Then the program crashes and they still are holding the memory. You think by logging out it will take care of the resource issue? What equates a active chat and a non active chat, is it a timed thing or based on when you logged into Fuckr last and who you contacted. There is no delete in the Inbox so I cannot remove some of these active chats. It is also curious that restarting does not get rid of them out of memory in "processes." When Fuckr crashes I often have to login and enter my password again, I suppose these phantom active chats do not see the crash and simply keep adding resources to new active chats used. 

I tried as you suggest to take the updated github forked files and install them by dragging into the Win 32 official version. When I did this it would not run and a coded screen appeared instead. 

![2016-02-06_023155](https://cloud.githubusercontent.com/assets/16879733/12866188/1fae98c0-cc7a-11e5-895a-d338e3654449.jpg)

**Comment by Kav28** [comment]
Theory that can be easily checked and it does mimic chrome behavior (more tabs more chrome processes).  Not sure how Windows works but in UNIX if a parent process dies child processes can be orphaned until the init process cleans them up and there can be zombies.  If the parent process exits on its own accord it should perform cleanup as part of that exit.  

Do you mean restarting your laptop or restarting fuckr?  The active chats are not every chat in the inbox (which is a history) but ones that started during the current fuckr session. Deleting old chats would have no effect.  

On Feb 6, 2016, at 2:34 AM, powline99 <notifications@github.com> wrote:

@Kav28 that could be a highly possible theory. So help me understand this. Every time I chat with multiple people on Fuckr and I do not logout there are forked resources which acts as basically active tabs in a web browser and use resources. Then the program crashes and they still are holding the memory. You think by logging out it will take care of the resource issue? What equates a active chat and a non active chat, is it a timed thing or based on when you logged into Fuckr last and who you contacted. There is no delete in the Inbox so I cannot remove some of these active chats. It is also curious that restarting does not get rid of them out of memory in "processes." When Fuckr crashes I often have to login and enter my password again, I suppose these phantom active chats do not see the crash and simply keep adding resources to new active chats used.

I tried as you suggest to take the updated github forked files and install them by dragging into the Win 32 official version. When I did this it would not run and a coded screen appeared instead.

**Comment by powline99** [comment]
@Kav28 That is the issue ..."a parent process dies child processes can be orphaned until the init process cleans them up and there can be zombies.  If the parent process exits on its own accord it should perform cleanup as part of that exit."

I have done a restart, and tried a shut down and power on. I often restart Fuckr without restarting because I am working on a bunch of other things and it crashes. I suppose I have alot of active chats going then, if they take up resources it would be nice to have them "suspend" like there is a plugin for chrome where they clear out until clicked on and made active again.

**Comment by powline99** [comment]
@victorgrego would you or anyone be so kind as to tell me how to use your updated fork (you updated the files so you can see more profile information, "specs"). I am using the 32 bit windows version from the official site. I download your zip from github and look at what files you updated, drag them into the folder was it layouts. I then startup fuckr.exe and there is no login screen. Must I wait for a official release and compile before using the updates or how does one use these forked updates with windows?

**Comment by victorgrego** [comment]
In my case i just replaced the files, the fuckr.js and the view. However this is a VERY developer version, and i found MANY bugs, you should wait for official release.

**Comment by powline99** [comment]
@victorgrego @tomlandia  Any updates on these features or a way to use @victorgrego Fork in a Windows version of the software?? What does "Delete disgusting profiles from profile view" mean I don't see any diffrence in the update. Thanks for you software!!!

- Advanced Filtering: Tribe, Relationship Status, Ethnicity, body type, looking for,
- Full stats on individual profiles like tribe, relationship status, orientation, ethnicity
- US Conversions Height - inches / Weight - pounds / Distance - Miles
- Darker / Minimal design like Scruff with no pink so I don't get in trouble using software to chat.
- Quicker way to block with keyboard shortcut so it does not take 5 steps from the tile view
- Way to Delete messages from inbox without blocking like Grindr has
- Emoji support, people think you are mentally challenged when you cannot understand the emoji icons you never saw, haha.
- Larger sent / shared pictures, there is no way to click and see them in a bigger size

**Comment by tomlandia** [comment]
Most of it in [v 1.6.0](https://github.com/tomlandia/fuckr/releases/1.6.0). For picture size, right now you can drag and drop. For a 'discrete' theme, I'm not sure yet... making Fuckr discrete would take more than just hiding the bar and changing colors. I'm leaving this open because of the Windows threading issue, any chance the Node-Webkit update fixed it?

**Comment by powline99** [comment]
@tomlandia Here is what happens. I am blocking or chatting away in the app, then it crashes, it then is stuck in memory using resources and I must restart the computer or it will not End Process. I updated to the newest Java / Flash and the same thing. So maybe your right it's a webkit thing or Grindr is crashing somehow and asking for Eclipse credentials.

**Comment by tomlandia** [comment]
Looks like https://groups.google.com/forum/#!topic/nwjs-general/LtJql3LSV44 may offer a solution provided it's stable and we can add a nw command line flag to releases

**Comment by victorgrego** [comment]
Tom, why not just make all pictures inside of a link tag. I would open inside of another blank screen in high quality. That was my naive solution. An adv. solution would be using bootstrap carousel to show all picture in a new window.

**Comment by powline99** [comment]
@tomlandia the newest version 1.6.1 still leaves multiple threads open in the processes after it crashes. It crashes quite often (i believe Grindr is logging me out and crashing possibly a eclipse login server issue - like if you have multiple instances open of your profile) on my Windows 32bit laptop and I must open up a new instance which takes more resources... 

![2016-04-28_082528](https://cloud.githubusercontent.com/assets/16879733/14893410/2af91562-0d24-11e6-8f77-cc0b97791d4f.jpg)

![2016-04-28_072714](https://cloud.githubusercontent.com/assets/16879733/14893412/30ca551e-0d24-11e6-9e16-dfcdcabd8554.jpg)

**Could we implement that "Looks like https://groups.google.com/forum/#!topic/nwjs-general/LtJql3LSV44 may offer a solution provided it's stable and we can add a nw command line flag to releases"**

**Comment by tomlandia** [comment]
Please try to limit a Github issue to one problem as it'd make maintenance easier.
